rust/gelato/src/sponsored_call.rs

```diff
@@ -0,0 +1,142 @@
+use crate::types::Chain;
+use crate::{types::serialize_as_decimal_str, RELAY_URL};
+use ethers::types::{Address, Bytes, U256};
+use serde::{Deserialize, Serialize};
+use tracing::instrument;
+
+#[derive(Debug, Clone, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SponsoredCallArgs {
+    pub chain_id: Chain,
+    pub target: Address,
+    pub data: Bytes,
+
+    // U256 by default serializes as a 0x-prefixed hexadecimal string.
+    // Gelato's API expects the gasLimit to be a decimal string.
+    #[serde(serialize_with = "serialize_as_decimal_str")]
+    pub gas_limit: Option<U256>,
+    pub retries: Option<u32>,
```

Comment:
Is there a difference between `Some(0)` and `None`?

Reply:
yeah the API will default to 5, adding this as a comment